CMakeLists.txt

```diff
@@ -1,39 +1,113 @@
 cmake_minimum_required(VERSION 2.8.3)
 project(openni2_tracker)
-find_package(catkin REQUIRED COMPONENTS geometry_msgs
-					orocos_kdl
+find_package(catkin REQUIRED COMPONENTS 
 					roscpp
 					roslib
 					tf)
-# Find OpenNI2
-#find_package(PkgConfig)
-#pkg_check_modules(OpenNI2 REQUIRED libopenni2)
-find_path(OpenNI2_INCLUDEDIR
-	  NAMES OpenNI.h
-	  HINTS /usr/include/openni2)
-find_library(OpenNI2_LIBRARIES
-	     NAMES OpenNI2 DummyDevice OniFile PS1090
-	     HINTS /usr/lib/ /usr/lib/OpenNI2/Drivers
-	     PATH_SUFFIXES lib) 
-message(STATUS ${OpenNI2_LIBRARIES})
-
-# Find Nite2
-message(status $ENV{NITE2_INCLUDE})
-message(status $ENV{NITE2_REDIST64})
-find_path(Nite2_INCLUDEDIR
-	  NAMES NiTE.h
-	  HINTS $ENV{NITE2_INCLUDE})
-find_library(Nite2_LIBRARY
-	     NAMES NiTE2
-	     HINTS $ENV{NITE2_REDIST64}
-	     PATH_SUFFIXES lib) 
+find_package(orocos_kdl REQUIRED)
 
 catkin_package()
 
-include_directories(${catkin_INCLUDEDIR}
-		    ${OpenNI2_INCLUDEDIR}
-		    ${Nite2_INCLUDEDIR})
+include(ExternalProject)
+## Check architecture 32 or 64 bit for precompiled OpenNI2 and NiTE2 Libraries
+if (CMAKE_SIZEOF_VOID_P MATCHES "8")
+	ExternalProject_Add(
+		OpenNI2
+		PREFIX ${CMAKE_CURRENT_BINARY_DIR}
+		URL http://www.openni.ru/wp-content/uploads/2013/01/OpenNI-Linux-x64-2.1.0.tar.zip
```

Comment:
This is not acceptable in my opinion, it should be downloaded from the structure.io website or better yet OpenNi's GitHub:

- http://structure.io/openni
- https://github.com/OpenNI/OpenNI/releases